subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -353,16 +349,16 @@ static int hawkbit_device_acid_update(int32_t new_value)
  */
 static void hawkbit_update_sleep(struct hawkbit_ctl_res *hawkbit_res)
 {
-	uint32_t sleep_time;
+	int sleep_time;
 	const char *sleep = hawkbit_res->config.polling.sleep;
 
 	if (strlen(sleep) != HAWKBIT_SLEEP_LENGTH) {
 		LOG_ERR("Invalid poll sleep: %s", sleep);
 	} else {
 		sleep_time = hawkbit_time2sec(sleep);
```

Comment:
not related to this PR but, from a quick look at `hawkbit_time2sec` implementation, it feels like there is probably a lot of room for improvement there in terms of error handling

Reply:
courtesy of chatgpt, might be a good starting point?

```
#include <stdlib.h> // For strtol
#include <stdbool.h> // For bool type

static int hawkbit_time2sec(const char *s) {
    long hours, minutes, seconds;
    char *next;

    // Validate and convert hours
    hours = strtol(s, &next, 10);
    if (*next != ':' || hours < 0) return -1;

    // Validate and convert minutes
    minutes = strtol(next + 1, &next, 10);
    if (*next != ':' || minutes < 0 || minutes >= 60) return -1;

    // Validate and convert seconds
    seconds = strtol(next + 1, &next, 10);
    if (*next != '\0' || seconds < 0 || seconds >= 60) return -1;

    return (hours * 3600) + (minutes * 60) + seconds;
}
```